contrib/platform/src/com/sun/jna/platform/win32/Wtsapi32.java

```diff
@@ -114,4 +178,65 @@
+     *            application is running.
+     * @param pLevel
+     *            A pointer to a DWORD variable that, on input, specifies the
+     *            type of information to return. To return an array of
+     *            WTS_PROCESS_INFO structures, specify zero. To return an array
+     *            of WTS_PROCESS_INFO_EX structures, specify one. If you do not
+     *            specify a valid value for this parameter, on output,
+     *            WTSEnumerateProcessesEx sets this parameter to one and returns
+     *            an error. Otherwise, on output, WTSEnumerateProcessesEx does
+     *            not change the value of this parameter.
+     * @param SessionID
+     *            The session for which to enumerate processes. To enumerate
+     *            processes for all sessions on the server, specify
+     *            WTS_ANY_SESSION.
+     * @param ppProcessInfo
+     *            A pointer to a variable that receives a pointer to an array of
+     *            WTS_PROCESS_INFO or WTS_PROCESS_INFO_EX structures. The type
+     *            of structure is determined by the value passed to the pLevel
+     *            parameter. Each structure in the array contains information
+     *            about an active process. When you have finished using the
+     *            array, free it by calling the WTSFreeMemoryEx function. You
+     *            should also set the pointer to NULL.
+     * @param pCount
+     *            A pointer to a variable that receives the number of structures
+     *            returned in the buffer referenced by the ppProcessInfo
+     *            parameter.
+     * @return If the function succeeds, the return value is a nonzero value. If
+     *         the function fails, the return value is zero. To get extended
+     *         error information, call the GetLastError function.
+     */
+    boolean WTSEnumerateProcessesEx(HANDLE hServer, IntByReference pLevel, DWORD SessionID,
+            PointerByReference ppProcessInfo, IntByReference pCount);
+
+    /**
+     * Frees memory that contains WTS_PROCESS_INFO_EX or WTS_SESSION_INFO_1
+     * structures allocated by a Remote Desktop Services function.
+     * 
+     * @param WTSTypeClass
+     *            A value of the WTS_TYPE_CLASS enumeration type that specifies
+     *            the type of structures contained in the buffer referenced by
+     *            the pMemory parameter.
+     * @param pMemory
+     *            A pointer to the buffer to free.
+     * @param NumberOfEntries
+     *            The number of elements in the buffer referenced by the pMemory
+     *            parameter.
+     * @return If the function succeeds, the return value is a nonzero value. If
+     *         the function fails, the return value is zero. To get extended
+     *         error information, call the GetLastError function.
+     */
+    boolean WTSFreeMemoryEx(int WTSTypeClass, Pointer pMemory, long NumberOfEntries);
```

Comment:
This should explode! `NumberOfEntries` is a `ULONG` and according to the MSDN that is 32 bit long and not 64bit:
https://docs.microsoft.com/en-us/windows/desktop/winprog/windows-data-types

Reply:
Good catch.  Interestingly, this was one of my first JNA mappings a few years ago, and I obviously made some errors then, most that I've fixed. But it's been in my project's code and hasn't caused a (known) problem yet.  Obviously needs fixing but may be a case where byte boundaries are saving my skin.

Reply:
I've been there and was burned by it - searching for something like this for a few hours sharpens sensitivity for this :-)